src/main/java/com/google/devtools/build/lib/remote/util/NetworkTime.java

```diff
@@ -87,9 +90,21 @@ public void start(Listener<RespT> responseListener, Metadata headers) {
           new ForwardingClientCallListener.SimpleForwardingClientCallListener<RespT>(
               responseListener) {
 
+            /**
+             * This method must not throw any exceptions. Doing so will cause the wrapped call to
+             * silently hang indefinitely: https://github.com/grpc/grpc-java/pull/6107
+             */
             @Override
             public void onClose(Status status, Metadata trailers) {
-              networkTime.stop();
+              // There is a risk that networkTime.stop() would throw a IllegalStateException: if
+              // networkTime.outstanding is overflowed, wallTime.stop() will be called even it's
+              // already stopped.
+              try {
+                networkTime.stop();
+                throw new IllegalStateException();
```

Comment:
Is this line debugging code left in? (otherwise how could this code end up not throwing an exception?)

Reply:
Ops, my bad, it was used for debugging. ( I told myself remember to delete that line)